product_packaging_level/models/product_packaging.py

```diff
@@ -28,15 +28,45 @@ class ProductPackaging(models.Model):
     qty_per_level = fields.Char(
         compute="_compute_qty_per_level", string="Qty per package level"
     )
+    name_type = fields.Selection(
+        string="Name Based On",
+        selection=[
+            ("by_package_level", "Package Level Name"),
+            ("by_package_type", "Package Type Name"),
+            ("user_defined", "Free text value in name field"),
```

Comment:
in my opinion, should replace "Free text value in name field" by "User Defined" or "Free Text"

Reply:
updated